python/tk_nuke_writenode/handler.py

```diff
@@ -812,6 +812,41 @@ def __get_publish_template(self, node, is_proxy=False):
         else:
             return self.__get_template(node, "publish_template")
     
+    def __get_used_output_names(self, node):
+        used_output_names = set()
+
+        # get the output names for all other nodes that are using the same profile
+        node_profile = self.get_node_profile_name(node)
+        for n in self.get_nodes():
+            if n != node and self.get_node_profile_name(n) == node_profile:
+                used_output_names.add(n.knob(TankWriteNodeHandler.OUTPUT_KNOB_NAME).value())
+
+        # get extra output names from file system if necessary
+        # ie if template doesn't include the "{version}" key
+        template = self.__get_render_template(node)
+        use_version = bool(template.keys.get("version"))
+
+        if not use_version:
+            output_key_name = None
+            for key_name in ["output", "channel"]:
+                if template.keys.get(key_name):
+                    output_key_name = key_name
+
+            fields = {}
```

Comment:
look like you have some duplicated code here with #12 , do you think you could factorize it?